src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -403,8 +404,8 @@ public UnfilteredPartitionIterator executeLocally(ReadExecutionController execut
         }
 
         Context context = Context.from(this);
-        UnfilteredPartitionIterator iterator = (null == searcher) ? Transformation.apply(queryStorage(cfs, executionController), new TrackingRowIterator(context))
-                                                                  : searchStorage(searcher, executionController);
+        UnfilteredPartitionIterator iterator = (null == searcher) ? Transformation.apply(queryStorage(cfs, executionController), new TrackingRowIterator(context, Type.READ_BYTES))
+                                                                  : Transformation.apply(searchStorage(searcher, executionController), new TrackingRowIterator(context, Type.SEARCH_BYTES));
```

Comment:
IIUIC, by doing so we're still tracking the bytes read from sstables as a result of the index query, but we're _not_ tracking the bytes read from the index itself: is that right? Did you consider/investigate what it would take to also track the index reads themselves?

Reply:
@sbtourist I took a pass on what it takes to track the actual read index bytes - the TLDR; it is not trivial and deserves a dedicated effort with deeper knowledge. I suggest we track the effort separate in light of our other sensor tracking priorities. 

Having said that, this PR still brings value:
* Index bytes are tracked
* Read bytes from tables with `indexQueryPlan` are now tracked (previously they would report 0 bytes in `READ_BYTES` sensor even for non-indexed cols. Also now, as a bare minimum, the index columns are accommodated for even if the user doesn't have them in the SELECT part by the nature of how `UnfilteredRowIterator` work and that fact the index search and the standard read combined are embedded together. 

Let me know if that makes sense: https://github.com/datastax/cassandra/pull/1019/commits/9da420f4da1062c25cd3f5b49bb1b5b3d12de754